core-blocks/paragraph/edit.js

```diff
@@ -0,0 +1,294 @@
+			// If before content is omitted, treat as intent to delete block.
+			onReplace( [] );
+		} else if ( content !== before ) {
+			// Only update content if it has in-fact changed. In case that user
+			// has created a new paragraph at end of an existing one, the value
+			// of before will be strictly equal to the current content.
+			setAttributes( { content: before } );
+		}
+	}
+
+	render() {
+		const {
+			attributes,
+			setAttributes,
+			mergeBlocks,
+			onReplace,
+			className,
+			backgroundColor,
+			textColor,
+			setBackgroundColor,
+			setTextColor,
+			fallbackBackgroundColor,
+			fallbackTextColor,
+			fallbackFontSize,
+		} = this.props;
+
+		const {
+			align,
+			content,
+			dropCap,
+			placeholder,
+		} = attributes;
+
+		const fontSize = this.getFontSize();
+
+		return (
+			<Fragment>
+				<BlockControls>
+					<AlignmentToolbar
+						value={ align }
+						onChange={ ( nextAlign ) => {
+							setAttributes( { align: nextAlign } );
+						} }
+					/>
+				</BlockControls>
+				<InspectorControls>
+					<PanelBody title={ __( 'Text Settings' ) } className="blocks-font-size">
+						<FontSizePicker
+							fontSizes={ FONT_SIZES }
+							fallbackFontSize={ fallbackFontSize }
+							value={ fontSize }
+							onChange={ this.setFontSize }
+						/>
+						<ToggleControl
+							label={ __( 'Drop Cap' ) }
+							checked={ !! dropCap }
+							onChange={ this.toggleDropCap }
+							help={ this.getDropCapHelp }
+						/>
+					</PanelBody>
+					<PanelColor
+						colorValue={ backgroundColor.value }
+						initialOpen={ false }
+						title={ __( 'Background Color' ) }
+						onChange={ setBackgroundColor }
+					/>
+					<PanelColor
+						colorValue={ textColor.value }
+						initialOpen={ false }
+						title={ __( 'Text Color' ) }
+						onChange={ setTextColor }
+					/>
+					<ContrastChecker
+						textColor={ textColor.value }
+						backgroundColor={ backgroundColor.value }
+						{ ...{
+							fontSize,
+							fallbackBackgroundColor,
+							fallbackTextColor,
+						} }
+					/>
+				</InspectorControls>
+				<RichText
+					tagName="p"
+					className={ classnames( 'wp-block-paragraph', className, {
+						'has-background': backgroundColor.value,
+						'has-drop-cap': dropCap,
+						[ backgroundColor.class ]: backgroundColor.class,
+						[ textColor.class ]: textColor.class,
+					} ) }
+					style={ {
+						backgroundColor: backgroundColor.value,
+						color: textColor.value,
+						fontSize: fontSize ? fontSize + 'px' : undefined,
+						textAlign: align,
+					} }
+					value={ content }
+					onChange={ ( nextContent ) => {
+						setAttributes( {
+							content: nextContent,
+						} );
+					} }
+					onSplit={ this.splitBlock }
+					onMerge={ mergeBlocks }
+					onReplace={ this.onReplace }
+					onRemove={ () => onReplace( [] ) }
+					placeholder={ placeholder || __( 'Add text or type / to add content' ) }
+				/>
+			</Fragment>
+		);
+	}
+}
+
+const ParagraphEdit = compose( [
```

Comment:
I noticed some differences withh the `master` branch. There were some changes applied to this code.  I will fix them myself.